pbsmetrics/metrics.go

```diff
@@ -0,0 +1,257 @@
+package pbsmetrics
+
+import (
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/rcrowley/go-metrics"
+	"github.com/vrischmann/go-metrics-influxdb"
+	"strings"
+	"time"
+)
+
+// Labels defines the labels that can be attached to the metrics.
+type Labels struct {
+	Source        DemandSource
+	RType         RequestType
+	Adapter       openrtb_ext.BidderName
+	PubID         string // exchange specific ID, so we cannot compile in values
+	Browser       Browser
+	CookieFlag    CookieFlag
+	RequestStatus RequestStatus
+}
+
+// Label typecasting. Se below the type definitions for possible values
+
+// DemandSource : Demand source enumeration
+type DemandSource string
+
+// RequestType : Request type enumeration
+type RequestType string
+
+// Browser type enumeration
+type Browser string
+
+// CookieFlag : User ID cookie exists flag
+type CookieFlag string
+
+// RequestStatus : The request/adapter return status
+type RequestStatus string
+
+// The demand sources
+const (
+	DemandWeb     DemandSource = "web"
+	DemandApp     DemandSource = "app"
+	DemandUnknown DemandSource = "unknown"
+)
+
+// The request types (endpoints)
+const (
+	ReqTypeLegacy RequestType = "legacy"
+	ReqTypeORTB2  RequestType = "openrtb2"
+	ReqTypeAMP    RequestType = "amp"
+)
+
+// Browser flag; at this point we only care about identifying Safari
+const (
+	BrowserSafari Browser = "safari"
+	BrowserOther  Browser = "other"
+)
+
+// Cookie flag
+const (
+	CookieFlagYes     CookieFlag = "2"
```

Comment:
Is there any reason these are numeric strings? If not, let's name them something real & readable.

`yes`, `no`, and `unknown` would do just fine... but since cookies have expiration dates, I might recommend renaming `yes` to `exists`. Just so that someday, if we find it useful, we can scale to `exists`, `expired`, `no`, and `unknown`.

(There may also be a better name for "no", but... I can't think of one offhand)

Reply:
Agreed.